code/functions/solution/functions.sol.cpp

```diff
@@ -0,0 +1,22 @@
+#include "Structs.h" // The data structs we will work with
+
+#include <cstdio> // For printing
+
+void printFiveCharacters(FastToCopy argument) {
+    printf("The first five characters are '%.5s'\n", argument.data);
+}
+
+void printFiveCharacters(const SlowToCopy & argument) {
+    //argument.data[0] = '\n' ;
```

Comment:
Is that line necessary?

Reply:
When uncommented, demonstrate that a try to modify argument will not compile.

Reply:
I added a comment to make the intention clearer.

Reply:
thx!